ros_tcp_endpoint/publisher.py

```diff
@@ -38,7 +37,12 @@ def __init__(self, topic, message_class, queue_size=10, latch=False):
         node_name = f"{strippedTopic}_RosPublisher"
         RosSender.__init__(self, node_name)
         self.msg = message_class()
-        self.pub = self.create_publisher(message_class, topic, queue_size)
+
+        qos_profile = QoSProfile(depth=queue_size)
+        if latch:
+            qos_profile.durability = QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL
```

Comment:
In tests, this didn't actually work unless I also set:

            qos_profile.reliability = QoSReliabilityPolicy.BEST_EFFORT
            qos_profile.history = QoSHistoryPolicy.KEEP_LAST

(I have no idea why the reliability setting would be necessary, but empirically that's what I found.)

It seems there's no simple way to translate the ROS1 concept of latching into ROS2 (c.f. https://github.com/ros2/ros2/issues/464) and trying to force it is just going to cause confusion. The correct fix is to support user-defined QoS settings.